### ISSUE TITLE
`clippy::cast_possible_truncation` does not warn when casting from `u128` to `usize`

### ISSUE THREAD

**Issue by xsrvmy** [rust-lang/rust-clippy#17764]
### Summary

The lint works for `u64` to `usize` but not `u128` to `usize`. From a quick glance at the code, it looks like at [line 113](https://github.com/rust-lang/rust-clippy/blob/df7957bde892aebe3d74c9e97f3b9f7d4a38f0f7/clippy_lints/src/casts/cast_possible_truncation.rs#L113) the cast-to-`usize` is hardcoded for `u64` only instead of anything above `u32`.

### Lint Name

clippy::cast_possible_truncation

### Reproducer

```
let x = u128::MAX;
let y = x as usize;
```
The second line should trigger the lint but does not.

### Version

```text
rustc 1.98.1 (48a229cea 2026-09-01)
binary: rustc
commit-hash: 48a229ceaefd4985c50990b14116b6d856af0985
commit-date: 2026-09-01
host: x86_64-pc-windows-msvc
release: 1.98.1
LLVM version: 22.1.8
```
